lib/grpc_kit/session/io.rb

```diff
@@ -52,8 +55,18 @@ def wait_readable
 
       # Blocking until io object is readable or writable
       # @return [void]
-      def select(timeout = 1)
-        ::IO.select([@io], [@io], [], timeout)
+      def select(timeout: 1, write: true)
+        rs, ws = ::IO.select([@io, @wake_o], write ? [@io] : [], [], timeout)
+        @wake_o.read(@wake_o.stat.size) if rs&.delete(@wake_o) && !@wake_o.closed?
+        [rs || [], ws || []]
+      end
+
+      # Wake thread blocked at #select method
+      def wake!(memo = nil)
```

Comment:
How about adding a comment that `memo` is used for a debugging purpose?

Reply:
```suggestion
      # @param [Symbol] Indicate what event needed to invoke blocking thread. This argument is for debugging purpose.
      def wake!(memo = nil)
```